### ISSUE TITLE
artifacts are stored in wrong places since 4.23.0

### ISSUE THREAD

**Issue by ck-trigfb** [firecow/gitlab-ci-local#249]
hi Mads,

i haven't had much time lately to play around with gitlab-ci-local. But today i updated to the newest version from a version somewhere in 4.19.x. I recognized some problems with the artifacts. Subsequent jobs can't access the artifacts, as it was possible before.

I have seen some change in the handling in .gitlab-ci-loca/ folder, now artifacts are stored with safeJobName as a subfolder instead of being stored directly in .gitlab-ci-loca/artifacts. So every job has its own subfolder lije
.gitlab-ci-loca/artifacts/build/path/to/deep/folder
.gitlab-ci-loca/artifacts/test/path/to/deep/folder
...

Another thing that might not be work as desired: The artifacts are also stored in the code repository. With your artifacts-docker testcase i get this result:
myCodeFolder/.gitlab-ci-local
myCodeFolder/.gitlab-ci-local/artifacts/build/path/to/deep/folder
myCodeFolder/path/to/deep/folder

The "path/to/deep/folder" folder is now inside of my code repository folder.

I now use version 4.22.0 as a workaround, so everything seems to work fine with that version

**Comment by firecow** [comment]
Do you have a mix of shell-executor and docker-executor jobs in your `.gitlab-ci.yml`...

If so try using the`--shell-isolation` option in gitlab-ci-local 4.24.0...

The "bugs" you are mention are in fact deliberate implementations. @ck-trigfb

**Comment by ck-trigfb** [comment]
Yes, i have a mix of shell-executor and docker-executer jobs. I don't know what i changed since yesterday, but now everything work fine again. Jobs can access artifacts from preceding jobs. So that's OK.

Can you explain, why artifacts are now saved into the root folder? In my opinion i use the gitlab pipeline to compile code inside a container and avoid by that way that all the build artifacts will pollute my local folder. Files and folder that are not already matched wit gitignore will even been added to the git repo.
 
In gitlab i have to download artifacts manually from each job and they will not automatically stored to the repository. At the moment i don't see the advantage of that behavior. Maybe you can add an option to deactivate on demand?

**Comment by firecow** [comment]
npm node_modules or php vendor folder artifacts are very nice to have around for your IDE...
A `@NoArtifactsToCwd` would be a good way of opt'ing out.

**Comment by bcouetil** [comment]
> npm node_modules or php vendor folder artifacts are very nice to have around for your IDE...

That seems a good idea at first, but we don't want gitlab to mess with our files, especially when having dozens of jobs producing artifacts. If we want PHP or whatever produced files, we can run the commands directly in our root folder !

Is it possible to get back the old behavior of artifacts only in `.gitlab-ci-local` subfolder ? :pray:

**Comment by firecow** [comment]
@bcouetil 
We cannot demand people have PHP installed directly on their machine in this day and age (read docker)

I'm facilitating many projects for hundreds of developers, if they have to install PHP5, PHP6 and PHP7 to run `composer install` in various projects, it will be a local maintenance hell, not to mention PECL extensions.

I need to make sure those IDE friendly artifacts (node_modules, vendor) can reach the developers source folder with them having only docker (k3 or minikube), bash and gitlab-ci-local installed on their local machine to run the projects.

I shall add `@NoArtifactsToSource`, and then we change it to `@ArtifactsToSource` in 5.x.x. 
That will give me some time to get `@ArtifactsToSource` descriptor copied to our CI modules files. :smile:

**Comment by bcouetil** [comment]
Not sur if this is something users are asking, but I understand the use case 😊

For now I'll do what it takes to not consider old artifacts.
